### PR TITLE
doc: update developer_guide.md

### PR DESCRIPTION
**Pull request by jialeif**
Update to kernel v6.2, qemu v7.2.0 and edk2-stable202302.

### CODE REVIEW

File: doc/developer_guide.md [hunk 1]
```diff
@@ -144,4 +142,4 @@ Please install these packages via distro's package manager.
    $ cp Build/IntelTdx/DEBUG_GCC*/FV/OVMF_CODE.fd /opt/tdx/tdvf/OVMF_CODE.debug.fd
    $ cp Build/IntelTdx/RELEASE_GCC*/FV/OVMF_CODE.fd /opt/tdx/tdvf/
    $ cp Build/IntelTdx/RELEASE_GCC*/FV/OVMF_VARS.fd /opt/tdx/tdvf/
-   ```
+`
```
Comment (ruomengh) [inline doc/developer_guide.md:145]: Extra ` needs to be removed

**Review by ruomengh — CHANGES_REQUESTED** [review]

File: doc/developer_guide.md (hunk 1 above)
Comment (jialeif) [reply doc/developer_guide.md]: Thanks!

**Review by kenplusplus — APPROVED** [review]
LGTM

File: doc/developer_guide.md [hunk 2]
```diff
@@ -20,9 +20,9 @@ Please install these packages via distro's package manager.
 2. Download the kernel source code:
 
    ```
-   $ wget https://github.com/intel/tdx-tools/blob/main/build/common/patches-tdx-kernel-MVP-KERNEL-5.19-v2.2.tar.gz
-   $ tar xf patches-tdx-kernel-MVP-KERNEL-5.19-v2.2.tar.gz
-   $ git clone --branch v5.19 https://github.com/torvalds/linux.git
+   $ wget https://github.com/intel/tdx-tools/blob/main/build/common/patches-tdx-kernel-MVP-KERNEL-6.2-v1.8.tar.z
```
Comment (ruomengh) [inline doc/developer_guide.md:23]: tar.z -> tar.gz?

**Review by ruomengh — CHANGES_REQUESTED** [review]

File: doc/developer_guide.md (hunk 2 above)
Comment (jialeif) [reply doc/developer_guide.md]: Thank you!

**Review by ruomengh — APPROVED** [review]
LGTM

**Pull request merged**